cvat-cli/src/cvat_cli/_internal/agent.py

```diff
@@ -192,21 +273,139 @@ def _validate_sublabel_compatibility(
+                    self._client.logger.info("Received notification about a new batch request")
+                    self._batch_request_might_be_available = True
+                    self._potential_work_condition.notify()
+                else:
+                    self._client.logger.warning(
+                        "Received notification about a new request of unknown category: %r",
+                        request_category,
+                    )
+        else:
+            self._client.logger.warning("Received event of unknown type: %r", event.type)
+
+    def _wait_before_reconnecting_to_queue(self):
+        delay_multiplier = self._rng.uniform(1, 1 + _JITTER_AMOUNT)
+        self._queue_watcher_should_stop.wait(
+            timeout=self._queue_reconnection_delay.total_seconds() * delay_multiplier
+        )
+
+        # Apply exponential backoff.
+        self._queue_reconnection_delay = min(
+            self._queue_reconnection_delay * 2, _POLLING_INTERVAL_MEAN_RARE
+        )
+
+    def _watch_queue(self) -> None:
+        while True:
+            # Until we can (re)connect to the event stream, poll more frequently.
+            self._polling_interval = _POLLING_INTERVAL_MEAN_FREQUENT
+
+            if self._queue_watcher_should_stop.is_set():
+                break
+
+            with self._queue_watch_response_lock:
+                self._client.logger.info("Attempting to watch the function's queue...")
+
+                try:
+                    _, self._queue_watch_response = self._client.api_client.call_api(
+                        "/api/functions/queues/{queue_id}/watch",
+                        "GET",
+                        path_params={"queue_id": f"function:{self._function_id}"},
+                        _parse_response=False,
+                    )
+                except Exception:
+                    self._client.logger.error(
+                        "Failed to connect to the queue event stream; will retry",
+                        exc_info=True,
+                    )
+                    self._wait_before_reconnecting_to_queue()
+                    continue
+                else:
+                    self._client.logger.info("Connected to the queue event stream")
+
+                    # Now we can rely on notifications, so slow down polling.
+                    self._polling_interval = _POLLING_INTERVAL_MEAN_RARE
+
+            try:
+                for event in _parse_event_stream(self._queue_watch_response):
+                    if isinstance(event, _Event):
+                        self._dispatch_queue_event(event)
+                    elif isinstance(event, _NewReconnectionDelay):
+                        self._queue_reconnection_delay = event.delay
+                        self._client.logger.info(
+                            "New queue event stream reconnection delay is %fs",
+                            self._queue_reconnection_delay.total_seconds(),
+                        )
```

Comment:
Consider adding a fallback for unknown event types, this will help to find errors if a new type is ever added.

Reply:
Sure, added.